openedx/core/djangoapps/safe_endpoints/tests/test_middleware.py

```diff
@@ -0,0 +1,77 @@
+"""
+Unit tests for safe_endpoints Middleware.
+"""
+import ddt
+
+from django.test import TestCase
+from rest_framework.authentication import SessionAuthentication
+from rest_framework_jwt.authentication import JSONWebTokenAuthentication
+from rest_framework.decorators import api_view
+from rest_framework.views import APIView
+
+from openedx.core.djangolib.testing.utils import get_mock_request
+
+from ..middleware import EnsureJWTAuthSettingsMiddleware
+
+
+class MockIncludedPermissionClass(object): pass
+class MockRequiredPermissionClass(object): pass
+
+
+def mock_auth_decorator(include_jwt_auth=True):
+    def _decorator(f):
+        f.permission_classes = (MockIncludedPermissionClass,)
+        f.authentication_classes = (SessionAuthentication,)
+        if include_jwt_auth:
+            f.authentication_classes += (JSONWebTokenAuthentication,)
+        return f
+    return _decorator
+
+
+@api_view(["GET"])
+@mock_auth_decorator(include_jwt_auth=False)
+def mock_function_view(request): pass
+
+
+@api_view(["GET"])
+@mock_auth_decorator(include_jwt_auth=True)
+def mock_function_view_with_jwt(request): pass
+
+
+@mock_auth_decorator(include_jwt_auth=False)
+class MockClassAPIView(APIView):
+    pass
+
+
+@mock_auth_decorator(include_jwt_auth=True)
+class MockClassAPIViewWithJwt(APIView):
+    pass
+
+
+@ddt.ddt
+class TestEnsureJWTAuthSettingsMiddleware(TestCase):
+    def setUp(self):
+        super(TestEnsureJWTAuthSettingsMiddleware, self).setUp()
+        self.request = get_mock_request()
+
+    @ddt.data(
+        (mock_function_view, False),
+        (mock_function_view_with_jwt, True),
```

Comment:
There are some missing combos.  Most important is probably having `permission_classes` not defined with authentication class of `JSONWebTokenAuthentication`.  You could also have a method with neither.

Reply:
Perhaps I'm misunderstanding your comment here.  The `include_jwt_auth` parameter indicates whether or not the `JSONWebTokenAuthentication` is specified by the view.

Reply:
To be more clear, your code under test includes the following line:
```
view_class.permission_classes = view_class.permission_classes or tuple()
```
But you don't test the condition where `view_class.permission_classes` is missing (if that is possible) or empty.  Would your code fail if `permission_classes` didn't exist at all?